examples/protocols/coap_server/main/coap_server_example_main.c

```diff
@@ -252,12 +252,16 @@ static void coap_example_server(void *p)
         }
 #if defined(CONFIG_COAP_MBEDTLS_PSK) || defined(CONFIG_COAP_MBEDTLS_PKI)
         if (coap_dtls_is_supported()) {
+#ifndef CONFIG_MBEDTLS_TLS_SERVER
+            ESP_LOGI(TAG, "MbedTLS (D)TLS Server Mode not configured");
```

Comment:
This can be changed to `ESP_LOGE` and add `goto clean_up`

Reply:
If not configured, the the server needs to support just plain coap:// and not fail - hence what I did

Reply:
I was thinking that we should not allow non-secure connections, but should be okay for this case as user would have explicitly disabled default configuration from mbedtls/coap.